### PR TITLE
Refactor plugin scanning into lib

### PR DESCRIPTION
**Pull request by pgomulka**
new stable plugins require generated named_components.json file which contains all analysis components implemented by this plugin. The generation is currently done in build-tools by elasticsearch.stable-esplugin
However this makes the generation only available for plugins using gradle. Plugin developers using maven or other building tooling will not be able to use it.

This commits refactors the scanning logic into libs:plugin-scanner which will allow for plugin install command to perform the scanning too.

relates https://github.com/elastic/elasticsearch/issues/88980

### CODE REVIEW

File: libs/plugin-scanner/build.gradle [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// This is only required because :server needs this at runtime.
+// We'll be removing this in 8.0 so for now just publish the JAR to make dependency resolution work.
```
Comment (breskeby) [inline libs/plugin-scanner/build.gradle:10]: is this comment actually true? I think this is outdated copied from somewhere?

Comment (breskeby) [reply libs/plugin-scanner/build.gradle:10]: I think we always need to publish this

File: libs/plugin-scanner/build.gradle [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+dependencies {
+  api project(':libs:elasticsearch-core')
+  api project(':libs:elasticsearch-plugin-api')
+  api project(":libs:elasticsearch-x-content")
+
+  implementation 'org.ow2.asm:asm:9.2'
+  implementation 'org.ow2.asm:asm-tree:9.2'
+//  implementation "com.fasterxml.jackson.core:jackson-core:${versions.jackson}"
```
Comment (breskeby) [inline libs/plugin-scanner/build.gradle:31]: can we clean that up

File: libs/plugin-scanner/build.gradle [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+// This is only required because :server needs this at runtime.
+// We'll be removing this in 8.0 so for now just publish the JAR to make dependency resolution work.
+apply plugin: 'elasticsearch.publish'
+
+tasks.named("test").configure { enabled = false }
+
+// test depend on ES core...
+tasks.named('forbiddenApisMain').configure { enabled = false}
```
Comment (breskeby) [inline libs/plugin-scanner/build.gradle:16]: why are those disabled?

File: libs/plugin-scanner/build.gradle [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// This is only required because :server needs this at runtime.
+// We'll be removing this in 8.0 so for now just publish the JAR to make dependency resolution work.
+apply plugin: 'elasticsearch.publish'
+
+tasks.named("test").configure { enabled = false }
```
Comment (breskeby) [inline libs/plugin-scanner/build.gradle:13]: that test task should be enabled as we actually have tests in that lib project

**Review by breskeby — CHANGES_REQUESTED** [review]
some minor changes. I think only having the tests enabled again is a blocker from my perspective to merge this though

File: libs/plugin-scanner/build.gradle (hunk 1 above)
Comment (pgomulka) [reply libs/plugin-scanner/build.gradle:10]: that was a copy paste. good spot

**Review by breskeby — APPROVED** [review]

**Pull request merged**